### PR TITLE
feat: add gradient clipping for pytorch scripts

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR adds gradient clipping for pytorch scripts (both detection & recognition)
Any feedback is welcome!

### CODE REVIEW

**Review by fg-mindee — APPROVED** [review]
Looks good to me!
